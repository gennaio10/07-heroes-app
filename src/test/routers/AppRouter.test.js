import React from "react";
import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { AuthContext } from "../../auth/AuthContext";
import { AppRouter } from "../../routers/AppRouter";

describe("Pruebas de <AppRouter />)", () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: { logged: false },
  };

  test("debe de mostrar el login si no esta autenticado", () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("button").text().trim()).toBe("Login");
  });

  test("debe de mostrar el componente de marvel si esta autenticado", () => {
    const contextValueScoped = {
      dispatch: jest.fn(),
      user: { logged: true, name: "Pororo" },
    };

    const wrapper = mount(
      <AuthContext.Provider value={contextValueScoped}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper.find("nav").exists()).toBe(true);
  });
});
