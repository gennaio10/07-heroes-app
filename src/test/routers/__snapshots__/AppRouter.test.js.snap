// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <AppRouter />) debe de mostrar el login si no esta autenticado 1`] = `
<div>
  <div
    className="container"
  >
    <br />
    <br />
    <br />
    <br />
    <br />
    <div
      className="d-flex justify-content-center h-100"
    >
      <div
        className="card"
      >
        <div
          className="card-header"
        >
          <h3>
            Sign In
          </h3>
        </div>
        <div
          className="card-body"
        >
          <div>
            <div
              className="input-group form-group"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                >
                  <i
                    className="fas fa-user"
                  />
                </span>
              </div>
              <input
                className="form-control"
                placeholder="username"
                type="text"
              />
            </div>
            <div
              className="input-group form-group"
            >
              <div
                className="input-group-prepend"
              >
                <span
                  className="input-group-text"
                >
                  <i
                    className="fas fa-key"
                  />
                </span>
              </div>
              <input
                className="form-control"
                placeholder="password"
                type="password"
              />
            </div>
            <div
              className="form-group"
            >
              <button
                className="btn btn-primary"
                onClick={[Function]}
              >
                Login
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
